### PR TITLE
Support signal polling

### PR DESCRIPTION
**Pull request by rado17**
Add support to handle signal polling requests from application.

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>

### CODE REVIEW

File: zephyr/src/supp_api.c [hunk 1]
```diff
@@ -280,6 +282,10 @@ int zephyr_supp_status(const struct device *dev,
 			/* TODO: Derive this based on association IEs */
 			status->link_mode = WIFI_6;
 		}
+		ret = wpa_drv_signal_poll(wpa_s, &si);
+		if (!ret) {
+			status->rssi = si.current_signal;
+		}
```
Comment (krish2718) [inline zephyr/src/supp_api.c:288]: log the failure?

File: src/drivers/driver_zephyr.c [hunk 2]
```diff
@@ -717,6 +717,28 @@ static int wpa_drv_zep_set_supp_port(void *priv,
...
 
 
+static int wpa_drv_zep_signal_poll(void *priv, struct wpa_signal_info *si)
+{
+	struct zep_drv_if_ctx *if_ctx = NULL;
+	const struct zep_wpa_supp_dev_ops *dev_ops = NULL;
+	int ret = -1;
+
+	if (!priv) {
+		wpa_printf(MSG_ERROR, "%s: Invalid handle\n", __func__);
```
Comment (krish2718) [inline src/drivers/driver_zephyr.c:727]: other funcs are using `Invalid params` and you can also check `!si`

File: src/drivers/driver_zephyr.c [hunk 3]
```diff
@@ -717,6 +717,28 @@ static int wpa_drv_zep_set_supp_port(void *priv,
...
+		goto out;
+	}
+
+	if_ctx = priv;
+	dev_ops = if_ctx->dev_ctx->config;
+
+	os_memset(si, 0, sizeof(*si));
+	ret = dev_ops->signal_poll(if_ctx->dev_priv, si, if_ctx->bssid);
```
Comment (krish2718) [inline src/drivers/driver_zephyr.c:735]: log error for failure?

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: src/drivers/driver_zephyr.c (hunk 3 above)
Comment (sachinthegreen) [inline src/drivers/driver_zephyr.c:735]: Shouldn't there be a check for if (!dev_ops->signal_poll) ?

File: zephyr/src/supp_api.c [hunk 4]
```diff
@@ -241,6 +241,8 @@ int zephyr_supp_status(const struct device *dev,
 				struct wifi_iface_status *status)
 {
 	struct wpa_supplicant *wpa_s;
+	struct wpa_signal_info si;
+	int ret;
```
Comment (sachinthegreen) [inline zephyr/src/supp_api.c:245]: Initialize to -1

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: zephyr/src/supp_api.c (hunk 4 above)
Comment (rado17) [reply zephyr/src/supp_api.c:245]: Initialized to -1

File: src/drivers/driver_zephyr.c (hunk 3 above)
Comment (rado17) [reply src/drivers/driver_zephyr.c:735]: Error is being checked and logged in the calling function.

Comment (krish2718) [reply src/drivers/driver_zephyr.c:735]: Typically, the errors should be logged the closes to the source, easier to debug.

File: src/drivers/driver_zephyr.c [hunk 5]
```diff
@@ -717,6 +717,43 @@ static int wpa_drv_zep_set_supp_port(void *priv,
...
+
+	if_ctx = priv;
+	dev_ops = if_ctx->dev_ctx->config;
+
+	os_memset(si, 0, sizeof(*si));
+
+	if (dev_ops && dev_ops->signal_poll) {
+		ret = dev_ops->signal_poll(if_ctx->dev_priv, si, if_ctx->bssid);
+		if (ret) {
+			wpa_printf(MSG_ERROR, "%s: Signal polling not supported\n", __func__);
```
Comment (krish2718) [inline src/drivers/driver_zephyr.c:744]: wring print, this should log the failure.

Comment (rado17) [reply src/drivers/driver_zephyr.c]: Corrected

File: zephyr/src/supp_api.c (hunk 4 above)
Comment (rado17) [reply zephyr/src/supp_api.c]: Done

File: src/drivers/driver_zephyr.c (hunk 3 above)
Comment (rado17) [reply src/drivers/driver_zephyr.c]: Added checks

Comment (rado17) [reply src/drivers/driver_zephyr.c]: Logged error

**Review by sachinthegreen — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]

**Pull request merged**